src/inbox/mod_inbox_muclight.erl

```diff
@@ -106,9 +106,17 @@ maybe_store_system_message(HostType, Room, Remote, Packet, Acc) ->
             ok
     end.
 
+-spec handle_room_destruction(HostType :: mongooseim:host_type(),
+                              Room :: jid:jid()) -> ok.
+handle_room_destruction(HostType, RoomJID) ->
+    CheckRemove = mod_inbox_utils:get_option_remove_on_kicked(HostType),
+    {ok, AffList, _} = mod_muc_light_db_backend:get_aff_users(HostType, jid:to_lus(RoomJID)),
+    [maybe_remove_inbox_row(HostType, RoomJID, UserJID, CheckRemove) || {UserJID, _Aff} <- AffList],
```

Comment:
Just a minor performance-related remark: we could check `CheckRemove` only once instead of doing it in a loop. But the impact is negligible, and we would need to change more code, so it's OK as it is for me.